docs/debug.md

```diff
@@ -0,0 +1,40 @@
+# Debugging the finch daemon
+
+This document outlines where to find/access logs and how to configure profiling tools for finch daemon.
+
+## Logs
+
+Logs are the first place to check when you suspect a problem with finch-daemon. If `finch-daemon` was started via `systemd` then you can obtain logs using `journalctl`:
+
+```shell
+sudo journalctl -u finch
+```
+
+> **Note**
+> The command above assumes that you have used the unit file definition [finch.service](../finch.service) we have provided. If you have created your own unit file for `finch-daemon` and replace `finch-daemon` with the one you have made. Amazon Linux distributions of Finch also use the name `finch` for the finch-daemon service.
+
+If you have started `finch-daemon` manually, logs will either be emitted to stderr/stdout.
+
+## CPU Profiling
+
+We can use Golangs `pprof` tool to profile the snapshotter. To enable profiling you must set the `--debug-addr` CLI parameter when invoking `finch-daemon`:
+
+```shell
+./finch-daemon --debug-addr localhost:6060
+```
+
+> **Note**
```

Comment:
Nit, may be helpful to just link https://pkg.go.dev/net/http/pprof to look for further details about how to generate and visualize different profiles.